oak_restricted_kernel/src/interrupts.rs

```diff
@@ -164,13 +164,164 @@ mutable_interrupt_handler_with_error_code!(
+    error!("KERNEL PANIC: X87 FLOATING POINT EXCEPTION!");
+    error!(
+        "Instruction pointer: {:#016x}",
+        stack_frame.deref().instruction_pointer.as_u64()
+    );
+    shutdown::shutdown();
+}
+
+extern "x86-interrupt" fn alignment_check_handler(
+    stack_frame: InterruptStackFrame,
+    error_code: u64,
+) {
+    error!("KERNEL PANIC: ALIGNMENT CHECK EXCEPTION!");
+    error!(
+        "Instruction pointer: {:#016x}",
+        stack_frame.deref().instruction_pointer.as_u64()
+    );
+    error!("Error code: {:?}", error_code);
+    shutdown::shutdown();
+}
+
+extern "x86-interrupt" fn machine_check_handler(stack_frame: InterruptStackFrame) -> ! {
+    error!("KERNEL PANIC: MACHINE CHECK EXCEPTION!");
+    error!(
+        "Instruction pointer: {:#016x}",
+        stack_frame.deref().instruction_pointer.as_u64()
+    );
+    shutdown::shutdown();
+}
+
+extern "x86-interrupt" fn simd_fp_handler(stack_frame: InterruptStackFrame) {
+    error!("KERNEL PANIC: SIMD FLOATING POINT EXCEPTION!");
+    error!(
+        "Instruction pointer: {:#016x}",
+        stack_frame.deref().instruction_pointer.as_u64()
+    );
+    shutdown::shutdown();
+}
+
 pub fn init_idt() {
     let mut idt = InterruptDescriptorTable::new();
+    idt.divide_error.set_handler_fn(divide_error_handler); // vector 0
+                                                           // skipping vector 1 (debug)
+    idt.non_maskable_interrupt.set_handler_fn(nmi_handler); // vector 2
     idt.breakpoint.set_handler_fn(breakpoint_handler); // vector 3
+    idt.overflow.set_handler_fn(overflow_handler); // vector 4
+    idt.bound_range_exceeded.set_handler_fn(bound_range_handler); // vector 5
+    idt.invalid_opcode.set_handler_fn(invalid_opcode_handler); // vector 6
+    idt.device_not_available
+        .set_handler_fn(device_not_available_handler); // vector 7
     idt.double_fault.set_handler_fn(double_fault_handler); // vector 8
+                                                           // vector 9 is reserved
+    idt.invalid_tss.set_handler_fn(invalid_tss_handler); // vector 10
+    idt.segment_not_present
+        .set_handler_fn(segment_not_present_handler); // vector 11
+    idt.stack_segment_fault
+        .set_handler_fn(stack_exception_handler); // vector 12
     idt.general_protection_fault
         .set_handler_fn(general_protection_fault_handler); // vector 13
     idt.page_fault.set_handler_fn(page_fault_handler); // vector 14
+                                                       // there is no vector 15
+    idt.x87_floating_point
+        .set_handler_fn(x87_floating_point_handler); // vector 16
+    idt.alignment_check.set_handler_fn(alignment_check_handler); // vector 17
+    idt.machine_check.set_handler_fn(machine_check_handler); // vector 18
+    idt.simd_floating_point.set_handler_fn(simd_fp_handler); // vector 19
+                                                             // there is no vector 20
```

Comment:
Is this a complete list of interrupts? Or only a subset?

Reply:
Only a subset. The Rust `IDT` struct exposes nice wrappers for only the first 20 exceptions; some we don't care about  (vector 21, which is not exposed by the IDT struct, is about shadow stacks which we do not have enabled, for example).

Even if we get an exception outside the range of the ones we're handling, we'd still get a double fault. And that'll tell us that the fault happened somewhere really weird.